gdal/frmts/jpegxr/jpegxrdataset.cpp

```diff
@@ -0,0 +1,1202 @@
+    /* -------------------------------------------------------------------- */
+    if (nBands != 1 && nBands != 3 && nBands != 4)
+    {
+        CPLError(CE_Failure, CPLE_NotSupported,
+            "JPEGXR driver doesn't support %d bands.  Must be 1 (grey), "
+            "3 (RGB) or 4 bands.\n", nBands);
+        return nullptr;
+    }
+
+    GDALDataType const eDT = poSrcDS->GetRasterBand(1)->GetRasterDataType();
+    if (eDT != GDT_Byte) // TODO
+    {
+        CPLError((bStrict) ? CE_Failure : CE_Warning, CPLE_NotSupported,
+            "JPEGCR driver doesn't support data type %s",
+            GDALGetDataTypeName(
+                poSrcDS->GetRasterBand(1)->GetRasterDataType()));
+
+        if (bStrict)
+            return nullptr;
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Collect configuration options                                   */
+    /* -------------------------------------------------------------------- */
+    JPEGXREncoderConfig config; // sets sensible defaults
+
+    if (CSLFetchNameValue(papszOptions, "QUALITY") != NULL)
+    {
+        int n = atoi(CSLFetchNameValue(papszOptions, "QUALITY"));
+        if (n < QUALITY_LOWEST || n > QUALITY_LOSSLESS)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                "QUALITY=%s is not a legal value in the range %d-%d.",
+                CSLFetchNameValue(papszOptions, "QUALITY"), QUALITY_LOWEST, QUALITY_LOSSLESS);
+            return nullptr;
+        }
+        config.SetQuality(n);
+    }
+
+    if (CSLFetchNameValue(papszOptions, "OVERLAP") != NULL)
+    {
+        int n = atoi(CSLFetchNameValue(papszOptions, "OVERLAP"));
+        if (OVERLAP_NONE < 0 || n > OVERLAP_COUNT)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                "OVERLAP=%s is not a legal value in the range %d-%d.",
+                CSLFetchNameValue(papszOptions, "OVERLAP"), OVERLAP_NONE, OVERLAP_COUNT-1);
+            return nullptr;
+        }
+        config.SetOverlap(n);
+    }
+
+    if (CSLFetchNameValue(papszOptions, "SUBSAMPLING") != NULL)
+    {
+        int n = atoi(CSLFetchNameValue(papszOptions, "SUBSAMPLING"));
+        if (SUBSAMPLING_420 < 0 || n > SUBSAMPLING_COUNT)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                "SUBSAMPLING=%s is not a legal value in the range %d-%d.",
+                CSLFetchNameValue(papszOptions, "SUBSAMPLING"), SUBSAMPLING_420, SUBSAMPLING_COUNT-1);
+            return nullptr;
+        }
+        config.SetSubsampling(n);
+    }
+    /* -------------------------------------------------------------------- */
+    /*      Read source data                                                */
+    /* -------------------------------------------------------------------- */
+    int nWordSize = GDALGetDataTypeSize(eDT) / 8;
+    int nUncompressedSize = nXSize * nYSize * nBands * nWordSize;
```

Comment:
If passed a big enough raster, this will int32 overflow. Is there a limit in dimension for JPEGXR files ? If so that could be checked before. Or if there's a limit of the uncompressed size (like 2 GB max), that could be checked here by casting to GIntBig and checking the result w.r.t INT_MAX.